src/sentry/api/event_search.py

```diff
@@ -579,17 +579,19 @@ def visit_boolean_filter(self, node, children):
         if self.is_numeric_key(search_key.name):
             return self._handle_numeric_filter(search_key, "=", [search_value.text, ""])
 
+        operator = "=" if not negated else "!="
+
         if self.is_boolean_key(search_key.name):
             if search_value.text.lower() in ("true", "1"):
-                search_value = SearchValue(0 if negated else 1)
+                search_value = SearchValue(1)
             elif search_value.text.lower() in ("false", "0"):
-                search_value = SearchValue(1 if negated else 0)
+                search_value = SearchValue(0)
             else:
                 raise InvalidSearchQuery(f"Invalid boolean field: {search_key}")
-            return SearchFilter(search_key, "=", search_value)
+            return SearchFilter(search_key, operator, search_value)
```

Comment:
Does this maybe break for if the column is empty?

Reply:
Before it would always be comparing to either True or False, now if you negate and say `!bool:true` it would produce the query `bool != 1`, which looking at how the `has` operator works, would include results which do not have this value.